website/server/controllers/api-v3/tasks/groups.js

```diff
@@ -16,6 +16,14 @@ import {
 
 let requiredGroupFields = '_id leader tasksOrder name';
 let types = Tasks.tasksTypes.map(type => `${type}s`);
+
+function canNotEditTasks(group, user, assignedUserId) {
+  let isNotGroupLeader = group.leader !== user._id;
+  let isManager = Boolean(group.managers[user._id]);
+  let userIsAssigningToSelf = Boolean(assignedUserId && user._id === assignedUserId);
+  let canNotEditTasks = isNotGroupLeader && !isManager && !userIsAssigningToSelf;
```

Comment:
Nope, this is not a change. If they are a group leader they can edit tasks. In English this says,

If you are not a group leader or a manager or assigning yourself, then you cannot assign this task.